src/components/backgroundselector/BackgroundService.js

```diff
@@ -214,20 +218,15 @@ goog.require('ga_glstylestorage_service');
 
         this.setById = function(map, newBgId) {
           if (map && (!bg || newBgId !== bg.id)) {
-            var newBg = getBgById(newBgId);
-            if (newBg) {
-              bg = newBg;
+            bg = getBgById(newBgId);
+            if (bg) {
               var layers = map.getLayers();
-              if (bg.id === 'voidLayer') {
-
+              if (bg.id === 'voidLayer' && layers.getLength() > 0 &&
+                  layers.item(0).background) {
```

Comment:
here you change the logic , if there is no layers on the map , you go in the 'else' and so create a layer. You don't want that

Reply:
yes correct